source/common/common/hash.h

```diff
@@ -25,6 +25,24 @@ class HashUtil {
     return XXH64(input.data(), input.size(), seed);
   }
 
+  /**
+   * Return 64-bit hash from the xxHash algorithm.
+   * @param input supplies the absl::InlinedVector<absl::string_view> to hash.
+   * @param seed supplies the hash seed which defaults to 0.
+   * See https://github.com/Cyan4973/xxHash for details.
+   */
+  static uint64_t xxHash64(absl::InlinedVector<absl::string_view, 1>& input, uint64_t seed = 0) {
+    uint64_t hash = seed;
+    if (input.size() == 0) {
+      hash = XXH64(input.data(), input.size(), hash);
```

Comment:
does this case occur?

If so, just use `XX64(nullptr, 0, seed)`

Reply:
Or maybe just don't test for that. Remove the `if` and let it return seed on an empty array.

Reply:
We didn't use that case when hashing the header value, special with the `num_headers_to_hash` `https://github.com/envoyproxy/envoy/pull/15486/files/ce1e9a7891114666566a8cb896bea0ca714758d6#diff-2830c8b332bc90cdc7f0e553551587f7ca8dafcf9c2565ab49e306b208909441R71`. 

I just match this behavior with `static uint64_t xxHash64(absl::string_view input, uint64_t seed = 0) `.
Then those two methods get the same value for empty string and empty vector, you can see that from the unittest
`https://github.com/envoyproxy/envoy/pull/15486/files/ce1e9a7891114666566a8cb896bea0ca714758d6#diff-a5b2aba566fe0501d4b46110c7fa658c92d2e5295e859adba66630bec15a4277L6-R20`

The upside to handle this empty vector case is in case someone pass the empty vector, then it will get a same behavior just like the empty string one.

But I'm also ok to remove it if you think it needless to a case we didn't use it today.